### PR TITLE
[ButtonBar] Increased KVO support for UIBarButtonItem properties.

### PR DESCRIPTION
**Pull request by jverkoey**
This allows button bar buttons to observe the following additional UIBarButtonItem properties:

 - accessibilityHint
 - accessibilityIdentifier
 - accessibilityLabel
 - accessibilityValue
- tag
- tintColor

Closes https://github.com/material-components/material-components-ios/issues/3692.

### CODE REVIEW

File: components/ButtonBar/src/MDCButtonBar.m
Comment (romoore) [inline components/ButtonBar/src/MDCButtonBar.m:312]: Nit: Extra newline here and below

File: components/ButtonBar/tests/unit/ButtonBarObservationTests.swift
Comment (romoore) [inline components/ButtonBar/tests/unit/ButtonBarObservationTests.swift]: Please provide *Given*, *When*, and *Then* comments for the different test sections.

File: components/ButtonBar/tests/unit/ButtonBarObservationTests.swift
Comment (romoore) [inline components/ButtonBar/tests/unit/ButtonBarObservationTests.swift]: Please make these separate tests (unless it's already covered elsewhere):
1. Initial values from UIButtonBarItem match MDCButton 
2. KVO changes are propagated

File: components/ButtonBar/tests/unit/ButtonBarObservationTests.swift
Comment (romoore) [inline components/ButtonBar/tests/unit/ButtonBarObservationTests.swift]: Super-nit: Prefer putting this method at the top so maintainers reading the file can see that the method exists locally (vs. globally) before it is used?

**Review by romoore — APPROVED** [review]

File: components/ButtonBar/src/MDCButtonBar.m
Comment (jverkoey) [reply components/ButtonBar/src/MDCButtonBar.m:312]: This is the format of this file already. The whitespace can be addressed in a follow-up change.
https://github.com/material-components/material-components-ios/issues/4381

File: components/ButtonBar/tests/unit/ButtonBarObservationTests.swift
Comment (jverkoey) [reply components/ButtonBar/tests/unit/ButtonBarObservationTests.swift]: Done.

File: components/ButtonBar/tests/unit/ButtonBarObservationTests.swift
Comment (jverkoey) [reply components/ButtonBar/tests/unit/ButtonBarObservationTests.swift]: Done.

**Comment by jverkoey** [comment]
Note that this PR is blocked by https://github.com/material-components/material-components-ios/pull/4379

File: components/ButtonBar/tests/unit/ButtonBarObservationTests.swift
Comment (jverkoey) [reply components/ButtonBar/tests/unit/ButtonBarObservationTests.swift]: Done.

File: components/ButtonBar/src/MDCButtonBar.m
Comment (romoore) [inline components/ButtonBar/src/MDCButtonBar.m:294]: Do you need to check for `respondsToSelector` as the code is doing for `setEnabled`?

File: components/ButtonBar/src/MDCButtonBar.m
Comment (romoore) [inline components/ButtonBar/src/MDCButtonBar.m:410]: Since these are used twice, please consider making them constants like `kEnabledSelector`

**Review by romoore — APPROVED** [review]

File: components/ButtonBar/src/MDCButtonBar.m
Comment (jverkoey) [reply components/ButtonBar/src/MDCButtonBar.m:294]: I don't believe so no.

File: components/ButtonBar/src/MDCButtonBar.m
Comment (jverkoey) [reply components/ButtonBar/src/MDCButtonBar.m:410]: kEnabledSelector was added as such because of compiler warnings - I'd prefer we keep referring to these APIs in a compile-time-safe way when possible, barring any definitive metrics that indicate we should do otherwise.

**Pull request merged**
